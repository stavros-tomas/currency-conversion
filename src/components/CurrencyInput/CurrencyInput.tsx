import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import InputLabel from '@mui/material/InputLabel';
import FormControl from '@mui/material/FormControl';
import TextField from '@mui/material/TextField';
import type { Currency } from '../../types';
import { ChangeEvent } from 'react';
import { SelectChangeEvent } from '@mui/material/Select';

interface CurrencyInputProps {
  label: string;
  currencies: Currency[];
  amount?: number | '';
  onCurrencyChange: (currency: string) => void;
  onAmountChange?: (amount: number) => void;
}

export default function CurrencyInput({ label, currencies, amount, onCurrencyChange, onAmountChange }: CurrencyInputProps) {
  const handleCurrencyChange = (event: SelectChangeEvent<string>) => {
    onCurrencyChange(event.target.value as string);
  };

  const handleAmountChange = (event: ChangeEvent<HTMLInputElement>) => {
    if (onAmountChange) {
      onAmountChange(Number(event.target.value));
    }
  };

  return (
    <FormControl className="tw-w-full">
      <div className="tw-flex tw-flex-row">
        <div className="tw-w-4/12">
          <InputLabel id={label}>{label}</InputLabel>
          <Select
            className="tw-w-full"
            defaultValue=''
            labelId={label}
            label={label}
            onChange={handleCurrencyChange}
            data-testid="currency-input__currency"
          >
            {currencies.length > 0 ? (
              currencies.map((currency) => (
                <MenuItem
                  key={`${label}-${currency.short_code}`}
                  value={currency.short_code}
                >
                  {currency.name}
                </MenuItem>
              ))
            ) : (
              <MenuItem disabled>No currencies available</MenuItem>
            )}
          </Select>
        </div>

        <div className="tw-w-8/12 tw-ml-2">
          <TextField
            value={amount}
            className="tw-w-full"
            type="number"
            id="filled-basic"
            label="Amount"
            onChange={handleAmountChange}
            data-testid="currency-input__amount"
          />
        </div>
      </div>
    </FormControl>
  );
}
