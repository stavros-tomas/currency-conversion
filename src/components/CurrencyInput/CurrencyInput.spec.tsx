import { render } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import CurrencyInput from './CurrencyInput';
import type { Currency } from '../../types';

const mockCurrencies: Currency[] = [
  { short_code: 'USD', name: 'United States Dollar'},
  { short_code: 'EUR', name: 'Euro'},
  { short_code: 'GBP', name: 'British Pound'},
];

describe('CurrencyInput Component', () => {
  it('should render correctly with given props', () => {
    const {container} =render(
      <CurrencyInput
        label="Currency"
        currencies={mockCurrencies}
        amount=""
        onCurrencyChange={vi.fn()}
        onAmountChange={vi.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
