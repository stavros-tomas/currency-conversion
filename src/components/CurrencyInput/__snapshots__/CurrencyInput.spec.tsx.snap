// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CurrencyInput Component > should render correctly with given props 1`] = `
<div>
  <div
    class="MuiFormControl-root tw-w-full css-1iw3t7y-MuiFormControl-root"
  >
    <div
      class="tw-flex tw-flex-row"
    >
      <div
        class="tw-w-4/12"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="Currency"
        >
          Currency
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl tw-w-full css-sc8y68-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          data-testid="currency-input__currency"
        >
          <div
            aria-controls=":r0:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="Currency"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="combobox"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </div>
          <input
            aria-hidden="true"
            aria-invalid="false"
            class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-81qg8w"
            >
              <span>
                Currency
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="tw-w-8/12 tw-ml-2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root tw-w-full css-1xp5r68-MuiFormControl-root-MuiTextField-root"
          data-testid="currency-input__amount"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="filled-basic"
            id="filled-basic-label"
          >
            Amount
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-quhxjy-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
              id="filled-basic"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-81qg8w"
              >
                <span>
                  Amount
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
